### PR TITLE
Added exception handling

### PR DESCRIPTION
**Pull request by distributive**
- Adds automatic error catching (currently only for command messages; not for inline commands or any other Discord events). Errors are displayed as embeds, but code remains for formatting them as plaintext messages if required.
- Merges `Tabletop/Util` into `Tabletop/Plugin` as I realised they were effectively serving the same purpose. Since I made that change `Tabletop/Util/Utils.hs` was added. I moved that to `Tabletop/Plugin/Utils.hs`.
- Removes void message-sending functions; void is now the default, and errors are caught and re-thrown as `MessageSendException`s.

### CODE REVIEW

File: tutorials/Plugins.md [hunk 1]
```diff
@@ -45,11 +45,11 @@ ping = Command "ping" (noArguments $ \m -> do
     return ())
 ```
 
-This can be further simplified through `sendMessageVoid`, which discards the error for us:
+This can be further simplified through `sendMessage`, which discards the error for us:
```
Comment (finnbar) [inline tutorials/Plugins.md:48]: This section needs more cut - the code example just above this no longer works.

Comment (distributive) [reply tutorials/Plugins.md]: I've updated this. If it's still a problem would you be able to commit your own rewrite?

File: tutorials/Exceptions.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+randomNumber =
+  Command
+    "randomNumber"
+    ( parseComm $ \m -> do
+        low <- number
+        space
+        high <- number
+        x <- chooseOne [low..high]
+        sendMessage m $ pack $ show x
+    )
```
Comment (finnbar) [inline tutorials/Exceptions.md:61]: does this work? it seems like you're merging parsing and commands here - perhaps use a smart command to get the numbers instead?

File: tutorials/Exceptions.md [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+```
+
+### Parser exceptions
+
+Be aware that you don't have to catch parsing errors yourself. The parser automatically throws a `ParserException` when it cannot match its input.
+
+## Outputting errors
+
+The exception module provides a number of ways to format your exceptions nicely. All BotExceptions implement the following functions:
```
Comment (finnbar) [inline tutorials/Exceptions.md:118]: > All BotExceptions implement the following functions

This implies that BotException is a typeclass which contains the following functions, which is isn't. Can just say "here are some functions for formatting your exceptions" or something

**Review by finnbar — CHANGES_REQUESTED** [review]
Some small tutorial fixes please! (otherwise the tutorial is great, and the rest of the code is solid too)

**Pull request merged**